### PR TITLE
Updated vagrant file to use qemu provider

### PR DESCRIPTION
**Pull request by pramitagautam**
Added qemu provider support to bring up the cluster

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -34,12 +34,20 @@ Vagrant.configure(2) do |config|
     controlplane.vm.box = "roboxes/ubuntu2004"
 
     controlplane.vm.network :private_network, ip:"#{k8s_linux_kubelet_nodeip}"
-    controlplane.vm.provider :virtualbox do |vb|
-    controlplane.vm.synced_folder "./sync/shared", "/var/sync/shared"
-    controlplane.vm.synced_folder "./forked", "/var/sync/forked"
-    controlplane.vm.synced_folder "./sync/linux", "/var/sync/linux"
-      vb.memory = linux_ram
```
Comment (knabben) [inline Vagrantfile:41]: do we need to keep the section for VirtualBox with the linux_ram and linux_cpus?

File: Vagrantfile [hunk 2]
```diff
@@ -62,10 +70,21 @@ Vagrant.configure(2) do |config|
     winw1.vm.box = "sig-windows-dev-tools/windows-2019"
     winw1.vm.box_version = "1.0"
 
-    winw1.vm.provider :virtualbox do |vb|
-      vb.memory = windows_ram
```
Comment (knabben) [inline Vagrantfile:66]: same here

File: Vagrantfile (hunk 1 above)
Comment (aravindhp) [reply Vagrantfile:41]: If possible we should support both until we get qemu fully working.

File: Vagrantfile [hunk 3]
```diff
@@ -34,12 +34,20 @@ Vagrant.configure(2) do |config|
...
-      vb.memory = linux_ram
-      vb.cpus = linux_cpus
+    controlplane.vm.synced_folder ".", "/vagrant", disabled: true
+    controlplane.vm.synced_folder "./sync/shared", "/var/sync/shared", type: "rsync"
+    controlplane.vm.synced_folder "./forked", "/var/sync/forked", type: "rsync"
+    controlplane.vm.synced_folder "./sync/linux", "/var/sync/linux", type: "rsync"
+
+    controlplane.vm.provider "qemu" do |qe|
+      qe.memory = linux_ram
+      qe.qemu_dir = "/opt/homebrew/share/qemu"
```
Comment (aravindhp) [inline Vagrantfile:44]: This directory is going to be different based on the base OS. Does it have a sane default based on the OS? IOW, do we need to specify it here?

Comment (pramitagautam) [reply Vagrantfile:44]: I remove and tested the VM bringup without this statement. And it works. So, should be ok to remove it.

**Comment by jayunit100** [comment]
lets just merge this to a QEMU branch and we'll iterate there

**Comment by jayunit100** [comment]
... lets merge to a qemu-branch and just hack there and then ill open a pr to merge it to main later this month

**Pull request closed without merging**